pkg/pipeline/resource_pipeline_node.go

```diff
@@ -0,0 +1,204 @@
+package pipeline
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type SystemPropertyBag struct {
+	Token string `json:"token"`
+}
+
+type Node struct {
+	//Project                   string          `json:"project"`
+	FriendlyName      string            `json:"friendlyName"`
+	ProjectId         int               `json:"projectId"`
+	NodePoolId        int               `json:"nodePoolId"`
+	IsOnDemand        bool              `json:"isOnDemand"`
+	IsAutoInitialized bool              `json:"isAutoInitialized"`
+	IPAddress         string            `json:"IPAddress,omitempty"`
+	IsSwapEnabled     bool              `json:"isSwapEnabled,omitempty"`
+	SystemPropertyBag SystemPropertyBag `json:"systemPropertyBag,omitempty"`
+	ID                int               `json:"id,omitempty"`
+}
+
+const nodesUrl = "pipelines/api/v1/nodes"
+
+func pipelineNodeResource() *schema.Resource {
+
+	var nodeSchema = map[string]*schema.Schema{
+		"friendly_name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+			Description:  "The name of the node. Should be prefixed with the project key",
+		},
+		"project_id": {
+			Type:         schema.TypeInt,
+			Required:     true,
+			ValidateFunc: validation.IntAtLeast(0),
```

Comment:
Are these numeric???? I swear a projects ID is a string (it's name) Like, 99% sure. @alexhung , since you implemented projects, what are your thoughts?

Reply:
This is the Pipeline's 'project' not Artifactory Project 😄 . According to API doc, they are integer.